process/blocksPool.go

```diff
@@ -1,46 +1,53 @@
 package process
 
 import (
+	"fmt"
 	"math"
 	"sync"
 
 	"github.com/multiversx/mx-chain-core-go/core"
 	"github.com/multiversx/mx-chain-core-go/core/check"
-	"github.com/multiversx/mx-chain-core-go/data/outport"
 	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-ws-connector-firehose-go/data"
 )
 
 const initIndex = 0
+const metaCheckpointKey = "lastMetaRound"
 
 type blocksPool struct {
-	storer          PruningStorer
-	marshaller      marshal.Marshalizer
-	maxDelta        uint64
-	cleanupInterval uint64
+	storer               PruningStorer
+	marshaller           marshal.Marshalizer
+	maxDelta             uint64
+	cleanupInterval      uint64
+	firstCommitableBlock uint64
+
+	previousIndexesMap map[uint32]uint64
+	mutMap             sync.RWMutex
+}
 
-	indexesMap map[uint32]uint64
-	mutMap     sync.RWMutex
+type BlocksPoolArgs struct {
+	Storer               PruningStorer
+	Marshaller           marshal.Marshalizer
+	MaxDelta             uint64
+	CleanupInterval      uint64
+	FirstCommitableBlock uint64
 }
 
 // NewBlocksPool will create a new blocks pool instance
-func NewBlocksPool(
-	storer PruningStorer,
-	marshaller marshal.Marshalizer,
-	maxDelta uint64,
-	cleanupInterval uint64,
-) (*blocksPool, error) {
-	if check.IfNil(storer) {
+func NewBlocksPool(args BlocksPoolArgs) (*blocksPool, error) {
+	if check.IfNil(args.Storer) {
 		return nil, ErrNilPruningStorer
 	}
-	if check.IfNil(marshaller) {
+	if check.IfNil(args.Marshaller) {
```

Comment:
should also add checks for uint64s against min/max values

Reply:
added additional checks